metomi/isodatetime/tests/test_00.py

```diff
@@ -1387,10 +1405,72 @@ def test_timerecurrence_comparison(self):
             test = hash(lhs) == hash(rhs)
             assert test is expected, (
                 "hash of {0} == hash of {1}".format(lhs_str, rhs_str))
+            # If recurrences the same, list of timepoints must be equal:
+            if lhs.repetitions is not None and rhs.repetitions is not None:
+                # Note: don't list() unbounded recurrences!
+                test = list(lhs) == list(rhs)
+                assert test is expected
         test_recurrence = parser.parse(tests[0][0])
         for var in [7, 'foo', (1, 2), data.Duration(days=1)]:
             self.assertFalse(test_recurrence == var)
 
+    def test_timerecurrence_add(self):
+        """Test adding/subtracting Duration to/from TimeRecurrence"""
+        rep = 4
+        start_pt = data.TimePoint(year=2020, month_of_year=3, day_of_month=13)
+        dur = data.Duration(days=7)
+
+        recurrence_fmt1 = data.TimeRecurrence(
+            repetitions=rep, start_point=start_pt,
+            end_point=data.TimePoint(year=2020, month_of_year=3,
+                                     day_of_month=20))
+        recurrence_fmt3 = data.TimeRecurrence(
+            repetitions=rep, start_point=start_pt, duration=dur)
+        recurrence_fmt4 = data.TimeRecurrence(
+            repetitions=rep, duration=dur,
+            end_point=data.TimePoint(year=2020, month_of_year=4,
+                                     day_of_month=3))
+        assert recurrence_fmt3 == recurrence_fmt4
```

Comment:
Yes, I figured as they are meant to be the same it would help to check that in case something breaks (although there is a separate test for the TimeRecurrence comparison operators, at least it should let the developer know it's not a problem with the add/sub methods).

Format 1 might have been missed out because I hadn't yet fixed that format when writing the test, not sure